deploy.sh

```diff
@@ -0,0 +1,130 @@
+# - the following environment variables
+#   - DEPLOY_DEST
+#   - WWW_ROOT # not used if not writing proxy/path config
+#   - USERDIR: the current user's home directory
+#   - PROXY_PORT: the port number being proxied (e.g. '4000')
+#   - PROXY_PATH: the path being proxied (e.g. '/api')
+#   - BASE_URL_PATH # not actually used but might be in future?
+#   - GLITCHTIP_KEY
+#   - MAPTILER_API_KEY
+#   - DBUS_SESSION_BUS_ADDRESS: required for service management
+# - FIXME the user can write to DEPLOY_DEST, WWW_ROOT
+#   - or else, the proxy pass and WWW roots have been defined already
+#
+# We don't assume a terminal or any interactivity.
+
+FE_DEST=$DEPLOY_DEST/front-end
+#BE_DEST=$DEPLOY_DEST/back-end
+BE_DEST=$PWD/apps/back-end # work around build glitches by running from source!
+DATA_DEST=$DEPLOY_DEST/data
+SYSTEMD_UNIT="$USERDIR/.config/systemd/user/mykomap-backend.service"
+[[ -z "$DBUS_SESSION_BUS_ADDRESS" ]] && {
+  die "no user session found? DBUS_SESSION_BUS_ADDRESS unset."
+}
+
+mkdir -vp "${DEPLOY_DEST:?}"
+
+
+mkdir -vp "${FE_DEST}" "${BE_DEST}" "${DATA_DEST}"
+
+
+# update asdf installations, using .tool-versions file
+asdf plugin add nodejs || true
+asdf install
+
+# Install systemd unit
+mkdir -p "${SYSTEMD_UNIT%/*}"
+cat >"$SYSTEMD_UNIT" <<EOF
+[Unit]
+Description=Mykomap back-end process manager for %u
+Documentation=https://github.com/DigitalCommons/mykomap-monolith/
+After=network.target
+
+[Service]
+Type=exec
+LimitNOFILE=infinity
+LimitNPROC=infinity
+LimitCORE=infinity
+Environment=PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin
+Environment=ASDF_DIR=/opt/asdf
+EnvironmentFile=$BE_DEST/.env
+Restart=on-failure
+WorkingDirectory=$BE_DEST
+ExecStart=bash -c '. "$ASDF_DIR/asdf.sh" && npm run start:attached'
+ExecReload=kill -HUP \$MAINPID
+
+[Install]
+WantedBy=default.target
+EOF
+
+# Keep ASDF happy when running in DEPLOY_DEST
+cp .tool-versions "$DEPLOY_DEST"
+
+( # Front end
+  cd apps/front-end
+  npm ci
+  rm -rf dist/
+
+  echo >.env
+  chmod 0600 .env # ensure secrets are secret-ish
+  # FIXME these shouldn't be hardwired!
+  cat >>.env <<EOF
+VITE_GLITCHTIP_KEY=${GLITCHTIP_KEY:?}
+VITE_MAPTILER_API_KEY=${MAPTILER_API_KEY:?}
+EOF
+  
+  npm run build
+  #npm deploy "$FE_DEST"
+  #cp -a --copy-contents dist/. "$FE_DEST" # the . is significant
+)
+( # back end
+  cd apps/back-end
+  npm ci
+  # this section doesn't work because of ESM/CJS glitches in the build
+  # rm -rf dist/
+  # npm run build 
+  # npm deploy "$BE_DEST"
+  # cp -a --copy-contents dist/. "$BE_DEST" # the . is significant
+
+  echo >"$BE_DEST/.env"
+  chmod 0600 "$BE_DEST/.env" # ensure secrets are secret-ish  
+  cat >>"$BE_DEST/.env" <<EOF
+SERVER_DATA_ROOT=$DATA_DEST
+FASTIFY_PORT=$PROXY_PORT
+#   root address?
+EOF
+  
+)
+
+# FIXME this needs perms! And root user ownership. Delegate to caller for now.
+false && {
```

Comment:
I delegated this stuff to the caller in the PBS install scripts (see the install.sh and install-remote.sh files)